packages/block-editor/src/components/rich-text/index.js

```diff
@@ -191,6 +187,19 @@ export function RichTextWrapper(
 					args: relatedBinding.args,
 				} );
 
+			// Don't modify placeholders if value is not empty.
+			if ( adjustedValue.length > 0 ) {
+				return {
+					disableBoundBlock: _disableBoundBlock,
```

Comment:
Interesting, should it explicitly return `null` for `bindingsPlaceholder` and `bindingsLabel` with some more detailed explanation? 

Alternatively, would the following work and made it easier to follow:

```suggestion
					disableBoundBlock: _disableBoundBlock,
					bindingsLabel: props[ 'aria-label' ] || placeholder,
```

Reply:
A similar trick might also work for `bindingsPlaceholder`.

Reply:
I've explicitly return `null` as suggested. I kind of like having the conditionals outside of the bindings logic. It makes it clear when the `placeholder` and the `aria-label` are defined:

* https://github.com/WordPress/gutenberg/blob/0027219426d4db3b7f5c38ee7eef844724024a35/packages/block-editor/src/components/rich-text/index.js#L372
* https://github.com/WordPress/gutenberg/blob/0027219426d4db3b7f5c38ee7eef844724024a35/packages/block-editor/src/components/rich-text/index.js#L442-L444

Having said that, I don't have a strong opinion so I'm happy to change that.